cmd/mdatagen/metrics_v2.tmpl

```diff
@@ -302,3 +302,8 @@ func (mb *MetricsBuilder) Reset(options ...metricBuilderOption) {
 		op(mb)
 	}
 }
+
+// SetStartTime sets the start time of metrics produced by the metrics builder to the given value.
+func (mb *MetricsBuilder) SetStartTime(startTime pcommon.Timestamp) {
+	mb.startTime = startTime
+}
```

Comment:
Please supply `WithStartTimeOverride` option to `EmitForResource` call instead of adding this method

Reply:
Done! This makes a lot more sense, I mistakenly thought it could only be used by the Emit function